### PR TITLE
refactor(cli): improve generation hints

### PR DESCRIPTION
**Pull request by millsp**
closes https://github.com/prisma/team-orm/issues/298

This PR changes how we display engine information during `prisma generate`. Three flavors:

`prisma generate` (default)

```sh
✔ Generated Prisma Client (v0.0.0) to ./generated/client in XXXms
```

`prisma generate` (binary engine)

```sh
✔ Generated Prisma Client (v0.0.0, engine=binary) to ./generated/client in XXXms
```

`prisma generate --no-engine`

```sh
✔ Generated Prisma Client (v0.0.0, engine=none) to ./generated/client in XXXms
```

Previously

```sh
✔ Generated Prisma Client (0.0.0 | library) to ./generated/client in XXXms
```

Also, I added something that should have been in a separate PR: we will now show a proper warning when URLs are hardcoded in the schema to let the user know this is a bad practice. This PR touches a lot of this area, so I opportunistically included it.

closes https://github.com/prisma/team-orm/issues/300

### CODE REVIEW

**Review by SevInf — APPROVED** [review]

File: packages/cli/src/generate/getHardcodedUrlWarning.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ConfigMetaFormat } from '@prisma/internals'
+
+export function getHardcodedUrlWarning(config: ConfigMetaFormat) {
+  if (config.datasources?.[0].provider !== 'sqlite' && config.datasources?.[0].url.fromEnvVar == null) {
+    return `\n🛑 Hardcoding URLs in your schema poses a security risk, use \`env("DATABASE_URL")\`\n`
```
Comment (jkomyno) [inline packages/cli/src/generate/getHardcodedUrlWarning.ts:5]:
```suggestion
    return `\n🛑 Hardcoding URLs in your schema poses a security risk, please use \`env("DATABASE_URL")\`\n`
```

Comment (jkomyno) [reply packages/cli/src/generate/getHardcodedUrlWarning.ts:5]: Do we have a `Learn more` docs page to link the users to?

**Review by jkomyno — APPROVED** [review]
Approved with a minor nit on the warning

**Comment by millsp** [comment]
@jkomyno I changed it to this, so the user can just open the docs now to read more about it

``sh
🛑 Hardcoding URLs in your schema poses a security risk: https://pris.ly/d/datasource-env
```

**Pull request merged**